packages/kbn-reporting/public/share/share_context_menu/register_pdf_png_modal_reporting.tsx

```diff
@@ -0,0 +1,505 @@
+                      <FormattedMessage
+                        id="reporting.share.publicNotifier.reportLink.reportingSectionUrlLinkLabel"
+                        defaultMessage="Stack Management &gt; Reporting"
+                      />
+                    </a>
+                  ),
+                }}
+              />,
+              { theme, i18n: i18nStart }
+            ),
+            'data-test-subj': 'queueReportSuccess',
+          });
+          if (onClose) {
+            onClose();
+          }
+        })
+        .catch((error: any) => {
+          toasts.addError(error, {
+            title: intl.formatMessage({
+              id: 'reporting.share.modalContent.notification.reportingErrorTitle',
+              defaultMessage: 'Unable to create report',
+            }),
+            toastMessage: error.body?.message,
+          });
+        });
+    };
+
+    shareActions.push({
+      shareMenuItem: {
+        name: i18n.translate('reporting.shareContextMenu.ExportsButtonLabel', {
+          defaultMessage: 'PDF',
+        }),
+        toolTipContent: licenseToolTipContent,
+        disabled: licenseDisabled || sharingData.reportingDisabled,
+        ['data-test-subj']: 'imageExports',
+      },
+      label: 'PDF' as const,
+      generateReportForPrinting: generateReportPDFForPrinting,
+      generateReport: generateReportPDF,
+      reportType: 'printablePdfV2',
+      requiresSavedState,
+      helpText: (
+        <FormattedMessage
+          id="reporting.printablePdfV2.helpText"
+          defaultMessage="Exports can take a few minutes to generate."
+        />
+      ),
+      generateReportButton: (
+        <FormattedMessage
+          id="reporting.printablePdfV2.generateButtonLabel"
+          defaultMessage="Generate export"
+        />
+      ),
+      layoutOption: objectType === 'dashboard' ? ('print' as const) : undefined,
+      theme,
+      renderLayoutOptionSwitch: objectType === 'dashboard',
+      renderCopyURLButton: true,
+    });
+
+    shareActions.push({
+      shareMenuItem: {
+        name: i18n.translate('reporting.shareContextMenu.ExportsButtonLabelPNG', {
+          defaultMessage: 'PNG export',
+        }),
+        toolTipContent: licenseToolTipContent,
+        disabled: licenseDisabled || sharingData.reportingDisabled,
+        ['data-test-subj']: 'imageExports',
+      },
+      label: 'PNG' as const,
+      generateReport: generateReportPNG,
+      reportType: 'pngV2',
+      requiresSavedState,
+      helpText: (
+        <FormattedMessage
+          id="reporting.pngV2.helpText"
+          defaultMessage="Exports can take a few minutes to generate."
+        />
+      ),
+      generateReportButton: (
+        <FormattedMessage
+          id="reporting.pngV2.generateButtonLabel"
+          defaultMessage="Generate export"
+        />
+      ),
+      layoutOption: objectType === 'dashboard' ? ('print' as const) : undefined,
+      renderCopyURLButton: true,
```

Comment:
The "POST URL" button is working for Discover/CSV, but on this screen it isn't actually copying the URL:
![test text to copy](https://github.com/elastic/kibana/assets/908371/145ecb78-7fbb-4a6f-a8bc-83cd9c48f91a)

Reply:
:face_palm: I forgot to add absoluteUrl for the register_pdf_png_modal. Fixed now thank you for the catch!